### PR TITLE
Support uploading file to empty repo by API

### LINKED ISSUE #14633: API: unable to post a first file to the empty repository
Repository was created with an API call [createCurrentUserRepo](http://localhost:3000/api/swagger#/repository/createCurrentUserRepo) and parameter "auto_init" set to false.

After that I want to create a first commit with README.md file, with the [repoCreateFile](http://localhost:3000/api/swagger#/repository/repoCreateFile) request:
`POST http://localhost:3000/api/v1/repos/lunik/apiRepoTest1/contents/README.md`
with the next body:
`{
  "content":"aGVsbG8==",
  "message": "initial"
}`

I've got the next response (also this is illegal JSON):

> {"message":"repo is empty","url":"http://localhost:3000/api/swagger"}
> {"message":"illegal base64 data at input byte 8","url":"http://localhost:3000/api/swagger"}

### PR DESCRIPTION
**Pull request by wxiaoguang**
The uploading API already works (the only nit is the the IsEmpty flag is out-of-sync, this PR also fixes it)

Close #14633

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: tests/integration/empty_repo_test.go [hunk 1]
```diff
@@ -105,3 +109,32 @@ func TestEmptyRepoUploadFile(t *testing.T) {
 	resp = session.MakeRequest(t, req, http.StatusOK)
 	assert.Contains(t, resp.Body.String(), "uploaded-file.txt")
 }
+
+func TestEmptyRepoAddFileByAPI(t *testing.T) {
+	defer tests.PrepareTestEnv(t)()
+
+	err := user_model.UpdateUserCols(db.DefaultContext, &user_model.User{ID: 30, ProhibitLogin: false}, "prohibit_login")
```
Comment (KN4CK3R) [inline tests/integration/empty_repo_test.go:116]: Can't you simply use another user?

Comment (wxiaoguang) [reply tests/integration/empty_repo_test.go:116]: I do not want to bloat the test fixtures. There are sooo many strange users/repos.

Comment (silverwind) [reply tests/integration/empty_repo_test.go:116]: It's good to keep tests self-contained in general, even if they are more repetive than app code.

**Review by silverwind — APPROVED** [review]

File: tests/integration/empty_repo_test.go (hunk 1 above)
Comment (wxiaoguang) [reply tests/integration/empty_repo_test.go:116]: user1 user2  are already used everywhere and a lot.

I do not think it's right to add user100 user999 user99999 in the future.

Comment (wxiaoguang) [reply tests/integration/empty_repo_test.go:116]: Actually, I have some plans to remove some unnecessary test fixtures, eg: the newly added user34 ......

Comment (silverwind) [reply tests/integration/empty_repo_test.go:116]: Could probably also add a shared user creation function that generated unique usernames (uuid etc.) with auto-assigned id.

Comment (silverwind) [reply tests/integration/empty_repo_test.go:116]: Yeah, and tests should clean up after themselves, e.g. delete created resources in both success and failure case 😆

Comment (wxiaoguang) [reply tests/integration/empty_repo_test.go:116]: Sure, ~~after each test~~ at the beginning of each test, all the data is reset, so this `UpdateUserCols` is 100% safe and right.

**Pull request merged**